src/Microsoft.AspNet.Mvc.ViewFeatures/Rendering/StringCollectionTextWriter.cs

```diff
@@ -41,12 +42,12 @@ public override Encoding Encoding
         /// A collection of entries buffered by this instance of <see cref="StringCollectionTextWriter"/>.
         /// </summary>
         // internal for testing purposes.
-        internal BufferEntryCollection Buffer { get; }
+        internal BufferedHtmlContent Content { get; }
 
         /// <inheritdoc />
         public override void Write(char value)
         {
-            Buffer.Add(value.ToString());
+            Content.Append(value.ToString());
```

Comment:
Yes, the reason why `char` is not present is because it is not a widely used overload in razor (`TagBuilder` uses it for adding some consts. I'm changing this in an upcoming PR). And the reason why `char[]` is not present is to avoid calling `TextWriter.Write()`s on each character and instead have it call `Write()` on the entire string once.
In general I'd prefer keeping `BufferedHtmlContent` simple; otherwise we'll have to worry about handling cases for each type we support. The only reason why it has a `List<object>` and not a `List<IHtmlContent>` is because we have to box strings everytime we store them (and strings are widely used in Razor).